src/sudoku.rs

```diff
@@ -279,14 +291,43 @@ mod tests {
         Row([1, 2, 3, 4]),
     ]);
 
+    #[test]
+    fn test_scorer() {
+        test_scorer_no_duplicates();
+        test_scorer_with_duplicates();
+    }
+
     #[test]
     fn test_board_fitness() {
         assert_eq!(0, GOOD_BOARD.fitness());
-        assert_eq!(12, BAD_BOARD.fitness());
+        assert_eq!(20, BAD_BOARD.fitness());
     }
 
     #[test]
     fn test_board_transpose() {
         assert_eq!(GOOD_BOARD_TRANSPOSED, GOOD_BOARD.transpose());
     }
+
+    fn test_scorer_no_duplicates() {
+        let mut scorer = Scorer::default();
+
+        // Since Scorer.seen is u64, it supports up to 49 before overflowing
```

Comment:
Since `Scorer.seen` is a `u64`, can we support up to 49 and update the `match` statement on line 163...or can we reduce this it to a `u32`?

I tested the following and saw it worked with `u32` without overflow, but does overflow `u16`:

```rust
        for i in 1..=25 {
            scorer.check(i);
        }
```